tests/TestCase/Cache/Engine/RedisClusterEngineTest.php

```diff
@@ -0,0 +1,283 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://book.cakephp.org/view/1196/Testing CakePHP(tm) Tests
+ * @since         5.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Test\TestCase\Cache\Engine;
+
+use Cake\Cache\Engine\RedisClusterEngine;
+use Cake\TestSuite\TestCase;
+use RedisCluster;
+use ReflectionClass;
+
+/**
+ * RedisClusterEngineTest class
+ */
+class RedisClusterEngineTest extends TestCase
+{
+    /**
+     * @var \Cake\Cache\Engine\RedisClusterEngine&\PHPUnit\Framework\MockObject\MockObject $engine
+     */
+    private RedisClusterEngine $engine;
+
+    /**
+     * setUp method
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->engine = $this->getMockBuilder(RedisClusterEngine::class)
+            ->onlyMethods(['connect'])
+            ->getMock();
```

Comment:
We would need to have tests that run against an actual redis cluster. Unfortunately as a framework we can't rely on mock tests as the real server's behavior is important too.